// angular-mgl-timeline/esm5/src/timeline/timeline-entry-side/timeline-entry-side.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, ElementRef, ViewEncapsulation } from '@angular/core';
var MglTimelineEntrySideComponent = /** @class */ (function () {
    function MglTimelineEntrySideComponent(elementRef) {
        this.elementRef = elementRef;
    }
    Object.defineProperty(MglTimelineEntrySideComponent.prototype, "alternate", {
        set: /**
         * @param {?} value
         * @return {?}
         */
        function (value) {
            this.elementRef.nativeElement.classList.toggle('alternate', value);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(MglTimelineEntrySideComponent.prototype, "mobile", {
        set: /**
         * @param {?} value
         * @return {?}
         */
        function (value) {
            this.elementRef.nativeElement.classList.toggle('mobile', value);
        },
        enumerable: true,
        configurable: true
    });
    MglTimelineEntrySideComponent.decorators = [
        { type: Component, args: [{
                    selector: 'mgl-timeline-entry-side',
                    template: "<ng-content></ng-content>",
                    encapsulation: ViewEncapsulation.None,
                    styles: ["mgl-timeline-entry-side{position:absolute;top:0;left:100%;width:100%;text-align:center}mgl-timeline-entry-side.alternate{left:-100%}mgl-timeline-entry-side.mobile{display:none}"]
                }] }
    ];
    /** @nocollapse */
    MglTimelineEntrySideComponent.ctorParameters = function () { return [
        { type: ElementRef }
    ]; };
    return MglTimelineEntrySideComponent;
}());
export { MglTimelineEntrySideComponent };
if (false) {
    /**
     * @type {?}
     * @private
     */
    MglTimelineEntrySideComponent.prototype.elementRef;
}
